src/transactiondust.php

```diff
@@ -0,0 +1,43 @@
+<?php
+namespace Ademti\Pwe;
+
+class TransactionDust {
+
+  /**
+   * The generated transaction dust amount.
+   *
+   * @var string
+   */
+  private $tx_dust;
+
+  /**
+   * Constructor.
+   *
+   * Generate and store a dust amount, or instantiate with the existing one.
+   *
+   * @param int $order_id  The order ID to generate a dust amount for.
+   */
+  public function __construct( $order_id ) {
+    $dust = get_post_meta( $order_id, '_pwe_dust_amount', true );
+    if ( $dust ) {
+      $this->dust_amount = $dust;
+    } else {
+      $this->dust_amount = substr(str_pad($order_id, 3, '0', STR_PAD_LEFT), -3);
```

Comment:
I thought that, but order ref is only going to possibly collide upon 999 successive orders, whereas random might be more often than that. We are limited to 8 decimal places if we want to enable coinbase mobile wallet payments.

I think that's a scenario which is unlikely enough that we can get away with this, otherwise we can remove some granularity from the order total and add a finer grained unique ID. Wdyt?

Reply:
Well, random in theory has the same collision likelihood (assume a suitable RNG), with the benefit that it makes it unpredictable. I don't care massively either way I don't think - just flagging.

Reply:
I think we'll go order number, makes me feel warmer inside. Feel free to change though, API doesn't care.